#PRINT PERMISSIVE MODULES
semanage permissive -l

#APPS !
semanage permissive -a zabbix_agent_t
semanage permissive -a zabbix_t
semanage permissive -a mysqld_t

#LOGROTATE
grep logrotate /var/log/audit/audit.log | audit2why

semanage permissive -a logrotate_t

#PRINT DATE FORMAT
ausearch -i

# TROUBLESHOOTING

grep $smtpd_t /var/log/audit/audit.log | audit2allow -M $NAME_SALIDA
grep zabbix_agentd /var/log/audit/audit.log | audit2allow -M zabbix2

audit2allow -a -M logrotate_t 
semodule -i logrotate_t.pp
chcon -R -t logrotate_t /[your log file location]/*.log

yum install policycoreutils-devel setroubleshoot-server
sealert -a /var/log/audit/audit.log

#SEARCH RECENT DENIES
ausearch -m avc --start recent

#AUDIT LOG MANAGE
# KH # 126101
https://www.thegeekdiary.com/how-to-stop-audit-log-entries-written-to-system-logs-in-centos-rhel-6/

[root@db02.scxca.prod.mgmt.sva.antel.com.uy ~]# grep args /etc/audisp/plugins.d/syslog.conf 
#args = LOG_INFO
args = LOG_LOCAL0

[root@db02.scxca.prod.mgmt.sva.antel.com.uy ~]# grep messages /etc/rsyslog.conf 
#*.info;mail.none;authpriv.none;cron.none -/var/log/messages
*.info;mail.none;authpriv.none;cron.none;local0.none /var/log/messages

[root@db02.scxca.prod.mgmt.sva.antel.com.uy ~]# service auditd restart
Stopping auditd:                                           [  OK  ]
Starting auditd:                                           [  OK  ]
[root@db02.scxca.prod.mgmt.sva.antel.com.uy ~]# service rsyslog restart
Shutting down system logger:                               [  OK  ]
Starting system logger:                                    [  OK  ]

#TRPUBLESHOOTER
[cl@cl ~]$ sudo ausearch -m AVC,USER_AVC -ts recent
<no matches>
[cl@cl ~]$ sudo journalctl -t setroubleshoot 
-- No entries --
[cl@cl ~]$ 

# OPEN VPN
#[cl@cl ~]$ sealert -l 3ccbbbce-5239-40b1-a535-f218dfdc3653
SELinux is preventing /usr/sbin/openvpn from open access on the file /root/.cert/nm-openvpn/pfsensep-UDP4-1194-clorenzo-config-2022-cert.pem.

*****  Plugin restorecon (99.5 confidence) suggests   ************************

If you want to fix the label. 
/root/.cert/nm-openvpn/pfsensep-UDP4-1194-clorenzo-config-2022-cert.pem default label should be home_cert_t.
Then you can run restorecon. The access attempt may have been stopped due to insufficient permissions to access a parent directory in which case try to change the following command accordingly.
Do
# /sbin/restorecon -v /root/.cert/nm-openvpn/pfsensep-UDP4-1194-clorenzo-config-2022-cert.pem

*****  Plugin catchall (1.49 confidence) suggests   **************************

If you believe that openvpn should be allowed open access on the pfsensep-UDP4-1194-clorenzo-config-2022-cert.pem file by default.
Then you should report this as a bug.
You can generate a local policy module to allow this access.
Do
allow this access for now by executing:
# ausearch -c 'openvpn' --raw | audit2allow -M my-openvpn
# semodule -X 300 -i my-openvpn.pp


Additional Information:
Source Context                system_u:system_r:openvpn_t:s0
Target Context                unconfined_u:object_r:admin_home_t:s0
Target Objects                /root/.cert/nm-openvpn/pfsensep-
                              UDP4-1194-clorenzo-config-2022-cert.pem [ file ]
Source                        openvpn
Source Path                   /usr/sbin/openvpn
Port                          <Unknown>
Host                          cl.kh
Source RPM Packages           openvpn-2.5.7-2.el9.x86_64
Target RPM Packages           
SELinux Policy RPM            selinux-policy-targeted-34.1.37-1.el9.noarch
Local Policy RPM              selinux-policy-targeted-34.1.37-1.el9.noarch
Selinux Enabled               True
Policy Type                   targeted
Enforcing Mode                Enforcing
Host Name                     cl.kh
Platform                      Linux cl.kh 5.14.0-130.el9.x86_64 #1 SMP
                              PREEMPT_DYNAMIC Fri Jul 15 11:24:09 UTC 2022
                              x86_64 x86_64
Alert Count                   4
First Seen                    2022-07-21 11:18:23 -03
Last Seen                     2022-07-21 11:29:20 -03
Local ID                      3ccbbbce-5239-40b1-a535-f218dfdc3653

Raw Audit Messages
type=AVC msg=audit(1658413760.268:398): avc:  denied  { open } for  pid=6023 comm="openvpn" path="/root/.cert/nm-openvpn/pfsensep-UDP4-1194-clorenzo-config-2022-cert.pem" dev="dm-0" ino=155322848 scontext=system_u:system_r:openvpn_t:s0 tcontext=unconfined_u:object_r:admin_home_t:s0 tclass=file permissive=0


type=SYSCALL msg=audit(1658413760.268:398): arch=x86_64 syscall=openat success=no exit=EACCES a0=ffffff9c a1=7ffecc74be3e a2=0 a3=0 items=0 ppid=6016 pid=6023 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm=openvpn exe=/usr/sbin/openvpn subj=system_u:system_r:openvpn_t:s0 key=(null)

Hash: openvpn,openvpn_t,admin_home_t,file,open

#https://ask.fedoraproject.org/t/selinux-is-preventing-openvpn-from-search-access-on-the-directory/11934
sudo fixfiles -F onboot
sudo systemctl reboot

